integration/basic_test.go

```diff
@@ -128,22 +128,74 @@ func TestFindCommentQuery(t *testing.T) {
 	assert.Contains(t, databaseNames, name)
 }
 
-func TestLongCollectionName(t *testing.T) {
+func TestCollectionName(t *testing.T) {
 	t.Parallel()
 
-	ctx, collection := Setup(t)
-
-	longCollectionName := "very_long_collection_name_that_is_more_than_64_characters_long_but_still_valid"
-	err := collection.Database().CreateCollection(ctx, longCollectionName)
-	require.NoError(t, err)
-
-	sixtyThreeCharsCollectionName := "this_is_a_collection_name_that_is_63_characters_long_abcdefghij"
-	err = collection.Database().CreateCollection(ctx, sixtyThreeCharsCollectionName)
-	require.NoError(t, err)
-
-	names, err := collection.Database().ListCollectionNames(ctx, bson.D{})
-	require.NoError(t, err)
-
-	assert.Contains(t, names, longCollectionName)
-	assert.Contains(t, names, sixtyThreeCharsCollectionName)
+	t.Run("Err", func(t *testing.T) {
+		ctx, collection := Setup(t)
+
+		cases := map[string]struct {
+			collection string
+			err        *mongo.CommandError
+			alt        string
+		}{
+			"TooLongForBoth": {
+				collection: "very_long_collection_name_that_fails_both_in_mongo_and_in_ferretdb_databases" +
+					"_for_ferretdb_it_fails_because_it_is_more_than_119_characters__for_mongo_it_fails_because_it_is_more_than_255_charachters_" +
+					"long_that_excludes_non_latin_letters_spaces_dots_dollars_dashes",
+				err: &mongo.CommandError{
+					Name: "InvalidNamespace",
+					Code: 73,
+					Message: "Fully qualified namespace is too long. Namespace: testcollectionname-err.very_long_collection_name_that_fails_both_in_mongo_" +
+						"and_in_ferretdb_databases_for_ferretdb_it_fails_because_it_is_more_than_119_characters__for_mongo_it_fails_because_it_is_more_than_" +
+						"255_charachters_long_that_excludes_non_latin_letters_spaces_dots_dollars_dashes Max: 255",
+				},
+				alt: "Collection must not contain non-latin letters, spaces, dots, dollars, dashes and be longer than 119 characters.",
+			},
+			"WithADollarSign": {
+				collection: "collection_name_with_a-$",
+				err: &mongo.CommandError{
+					Name:    "InvalidNamespace",
+					Code:    73,
+					Message: `Invalid collection name: collection_name_with_a-$`,
+				},
+				alt: "Collection must not contain non-latin letters, spaces, dots, dollars, dashes and be longer than 119 characters.",
```

Comment:
Can we return the same error message there and in other places?

Reply:
> Can we return the same error message there and in other places?

there -> where?
other places - > which places do you mean?

Reply:
As we discussed previously, we should return the same error message as MongoDB where we can. I this test case MongoDB returns `Invalid collection name: <collection name>`. That's a good correct error message. It should not be hard to make FerretDB return the same error message. So let's do that here and in other places in this PR.

Reply:
Let's keep sentinel error `ErrInvalidTableName` as is for now then, and not add a new type